data/model/autoprune.py

```
@@ -514,15 +515,40 @@ def create_autoprune_task(namespace_id):
+    Retries up to 3 times with exponential backoff when PostgreSQL deadlocks occur.
+    Total of 4 attempts (1 initial + 3 retries) with backoff delays of 0.1s, 0.2s, 0.4s.
+    """
+    max_retries = 3  # number of retries after the initial attempt
+    for attempt in range(max_retries + 1):
+        try:
+            task.status = task_status
+            task.save()
+            return
+        except AutoPruneTaskStatus.DoesNotExist:
+            return None
+        except Exception as err:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)